http-client/src/test/java/io/airlift/http/client/TestGatheringByteArrayInputStream.java

```diff
@@ -0,0 +1,156 @@
+package io.airlift.http.client;
+
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static java.lang.Math.min;
+import static java.lang.System.arraycopy;
+import static org.testng.Assert.assertEquals;
+
+public class TestGatheringByteArrayInputStream
+{
+    @Test
+    public void testNormal()
+    {
+        byte[] lastStringArray = "client".getBytes();
+        byte[] lastArray = new byte[lastStringArray.length + 32];
+        arraycopy(lastStringArray, 0, lastArray, 0, lastStringArray.length);
```

Comment:
This can be `byte[] lastArray = Arrays.copyOf(lastStringArray, lastStringArray.length + 32`